File: cai/118.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if not numRows:
            return []
        out=[[1]]
        while numRows>1:
            out=out+[[1]+[out[-1][i]+out[-1][i+1] for i in range(len(out[-1])-1)]+[1]]
            numRows=numRows-1
        return out

File: xiong/118.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        if(numRows == 0) return vector<vector<int>>{};
        vector<int> first = {1};
        if(numRows == 1) return vector<vector<int>>{first};
        vector<int> sec   = {1, 1};
        if(numRows == 2) return vector<vector<int>>{first, sec};
        vector<vector<int>> ans = {first, sec};
        for(int i = 0; i < numRows - 2; ++i){
            vector<int> now = {};
            vector<int> b = ans.back();
            now.push_back(1);
            for(int j = 0; j < b.size() - 1; ++j){
                now.push_back(b[j] + b[j + 1]);
            }
            now.push_back(1);
            ans.push_back(now);
        }
        return ans;
        
    }
};